src/vscode-dts/vscode.proposed.toolProgress.d.ts

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	/**
+	 * todo@connor4312: `vscode.window.withProgres` can take this interface as well.
+	 */
+	export interface ProgressStep {
```

Comment:
Unsure about the name (and about resuing this for `withProgress`). The idea of `Progress<T>` is that `T` is specific to the place/domain of progress. So, maybe `ToolProgressStep` or keep it inlined